cpp/ql/src/semmle/code/cpp/Function.qll

```diff
@@ -943,9 +1054,18 @@ class CopyConstructor extends Constructor {
 }
 
 /**
- * A C++ move constructor, such as `T::T(T&&)` [N4140 12.8].
+ * A C++ move constructor [N4140 12.8]. For example the function `MyClass` in
+ * the following code is a `MoveConstructor`:
+ * ```
+ * class MyClass {
+ * public:
+ *   MyClass(const MyClass &&from) {
```

Comment:
There should not be `const` here. See https://en.cppreference.com/w/cpp/language/move_constructor.

Reply:
I just had a second look at this, and I can see that it's technically legal to have `const` there, but I'd still argue that we should remove the `const`. It makes little sense to declare an rvalue reference as const since the whole purpose is to move out of it, which will lead to modifying it in non-trivial cases.

Reply:
I haven't really played with move constructors or thought about them much, just copied the syntax I'd seen elsewhere (which probably did have a non-const version alongside).  I've removed `const`.